lib/chromedriver.js

```diff
@@ -513,71 +520,36 @@ export class Chromedriver extends events.EventEmitter {
 
   /**
    * Sync the WebDriver protocol if current on going protocol is W3C or MJSONWP.
-   * Does nothing if this.driverVersion is null.
+   * Does nothing if driverVersion is null.
    *
-   * @returns {typeof PROTOCOLS[keyof typeof PROTOCOLS]}
+   * @returns {keyof PROTOCOLS}
    */
   syncProtocol() {
-    if (!this.driverVersion) {
-      // Keep the default protocol if the driverVersion was unsure.
-      return this.desiredProtocol;
+    if (this.driverVersion) {
+      const coercedVersion = semver.coerce(this.driverVersion);
+      if (!coercedVersion || coercedVersion.major < MIN_CD_VERSION_WITH_W3C_SUPPORT) {
+        this.log.info(
+          `The ChromeDriver v. ${this.driverVersion} does not fully support ${PROTOCOLS.W3C} protocol. ` +
+            `Defaulting to ${PROTOCOLS.MJSONWP}`,
+        );
+        this._desiredProtocol = PROTOCOLS.MJSONWP;
+        return this._desiredProtocol;
+      }
     }
 
-    this.desiredProtocol = PROTOCOLS.MJSONWP;
-    const coercedVersion = semver.coerce(this.driverVersion);
-    if (!coercedVersion || coercedVersion.major < MIN_CD_VERSION_WITH_W3C_SUPPORT) {
-      this.log.info(
-        `The ChromeDriver v. ${this.driverVersion} does not fully support ${PROTOCOLS.W3C} protocol. ` +
-          `Defaulting to ${PROTOCOLS.MJSONWP}`,
-      );
-      return this.desiredProtocol;
-    }
     // Check only chromeOptions for now.
     const chromeOptions = getCapValue(this.capabilities, 'chromeOptions', {});
     if (chromeOptions.w3c === false) {
       this.log.info(
         `The ChromeDriver v. ${this.driverVersion} supports ${PROTOCOLS.W3C} protocol, ` +
           `but ${PROTOCOLS.MJSONWP} one has been explicitly requested`,
       );
-      return this.desiredProtocol;
+      this._desiredProtocol = PROTOCOLS.MJSONWP;
```

Comment:
`chromeOptions.w3c === false` this condition could occur in chrome/msedge drivers. In operadriver case, even we don't provide `chromeOptions.w3c`, it works as non-w3c. So only when explicitly set `chromeOptions.w3c` as `true`, operadriver behaves as W3C.


This behavior is the same as chromedriver 131 based operadriver as well, so, probably in current logc, operadriver will be handled as W3C.

In current master's case, probably operadriver was matched with `// Keep the default protocol if the driverVersion was unsure.` so the desiredProtocl was `PROTOCOLS.MJSONWP`

I'll take a look further tonight

Reply:
that is why the change is marked as breaking.
Generally, we would like to always default to W3C as JWP has been deprecated several years ago.

Reply:
Yea, agree with that (to make w3c default)